src/vector/backends/object.py

```diff
@@ -799,17 +830,47 @@ class MomentumObject2D(PlanarMomentum, VectorObject2D):
     """
     Two dimensional momentum vector class for the object backend.
 
+    Examples:
+        >>> import vector
+        >>> vec = vector.MomentumObject2D(px=1, py=2)
+        >>> vec.px, vec.py
+        (1, 2)
+        >>> vec = vector.MomentumObject2D(pt=1, phi=2)
+        >>> vec.pt, vec.phi
+        (1, 2)
+        >>> vec = vector.MomentumObject2D(azimuthal=vector.backends.object.AzimuthalObjectXY(1, 2))
+        >>> vec.px, vec.py
+        (1, 2)
+
+    The :func:`vector.obj` function can also be
+    used to construct 2D momentum object type vectors.
+
     For two dimensional vector objects, see
     :class:`vector.backends.object.VectorObject2D`.
     """
 
+    def __init__(
+        self, azimuthal: typing.Optional[AzimuthalObject] = None, **kwargs: float
+    ) -> None:
+        if not kwargs and azimuthal is not None:
+            self.azimuthal = azimuthal
+        elif kwargs and azimuthal is None:
+            if set(kwargs) == {"px", "py"}:
+                self.azimuthal = AzimuthalObjectXY(kwargs["px"], kwargs["py"])
+            elif set(kwargs) == {"pt", "phi"}:
+                self.azimuthal = AzimuthalObjectRhoPhi(kwargs["pt"], kwargs["phi"])
+            else:
+                raise TypeError("invalid arguments, must be px=, py= or pt=, phi=")
+        else:
+            raise TypeError("must give Azimuthal if not giving keyword arguments")
+
```

Comment:
I went through the overloads of `vector.obj` and a user can indeed create a `MomentumObject` by specifying a mixture of momentum and generic coordinates. I think this should be removed and the momentum coordinate names must be converted to generic in the `__init__` method of `VectorObject` classes, which will then be inherited by `MomentumObject` classes.